docs/frameworks/nestjs.md

```diff
@@ -0,0 +1,161 @@
+---
+title: NestJS
+description: Learn how to integrate Featurevisor in Nest.js applications
+ogImage: /img/og/docs-frameworks-nest.png
+---
+
+Set up Featurevisor SDK instance in a Nest.js application using a custom middleware, including TypeScript integration. {% .lead %}
+
+## Hello World application
+
+Before going into Featurevisor integration, let's create a simple Hello World [Nest.js](https://nestjs.com/) application.
+
+We start by installing the Nest CLI:
+
+```bash
+$ npm install -g @nestjs/cli
+```
+
+Then we create a new Nest.js project:
+
+```bash
+$ nest new featurevisor-nest-app
+$ cd featurevisor-nest-app
+```
+
+## Featurevisor integration
+
+We install the Featurevisor SDK first:
+
+```bash
+$ npm install --save @featurevisor/sdk
+```
+
+We can now create an instance of the SDK and use it in our application:
+
+```typescript
+// main.ts
+import { NestFactory } from '@nestjs/core';
+import { AppModule } from './app.module';
+import { createInstance } from '@featurevisor/sdk';
+
+// Define constants
+const DATAFILE_URL = 'https://cdn.yoursite.com/datafile.json';
+const REFRESH_INTERVAL = 60 * 5; // every 5 minutes
+
+async function bootstrap() {
+  const app = await NestFactory.create(AppModule);
+
+  const f = createInstance({
+    datafileUrl: DATAFILE_URL,
+    refreshInterval: REFRESH_INTERVAL,
+  });
+
+  app.use((req, res, next) => {
+    req.f = f;
+    next();
+  });
+
+  await app.listen(3000);
+}
+bootstrap();
+```
+
+## middleware
+
+To use the same Featurevisor SDK instance in all of our routes, we can create a custom middleware in Nest.js.
+
+Create a new file featurevisor.middleware.ts:
+
+```typescript
+// featurevisor.middleware.ts
+import { Injectable, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { FeaturevisorInstance, createInstance } from '@featurevisor/sdk';
+
+@Injectable()
+export class FeaturevisorMiddleware implements NestMiddleware {
+  private readonly f: FeaturevisorInstance;
+
+  constructor() {
+    const DATAFILE_URL = 'https://cdn.yoursite.com/datafile.json';
+    const REFRESH_INTERVAL = 60 * 5; // every 5 minutes
+
+    this.f = createInstance({
+      datafileUrl: DATAFILE_URL,
+      refreshInterval: REFRESH_INTERVAL,
+    });
+  }
+
+  use(req: Request, res: Response, next: NextFunction) {
+    req.f = this.f;
+
+    next();
+  }
+}
+```
+
+And then update the main.ts file to use the middleware:
+
+```typescript
+// main.ts
+import { NestFactory } from '@nestjs/core';
+import { AppModule } from './app.module';
+import { FeaturevisorMiddleware } from './featurevisor.middleware';
+
+async function bootstrap() {
+  const app = await NestFactory.create(AppModule);
+
+  app.use(new FeaturevisorMiddleware().use);
```

Comment:
I am curious, is there any added benefit of creating a new `FeaturevisorMiddlleware` class with a `use` method, and then instantiate it only to pass its `use` method directly to `app.use()`?

we could have also done it by just passing it as a function from here.